ImageSaver/app/src/main/java/com/example/jong/imagesaver/Image_query.java

```diff
@@ -0,0 +1,195 @@
+        switch(button){
+            case("save"):{
+                Button Save;
+                Save = (Button) v.findViewById(R.id.save_button);
+                Save.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        String imgurl = listurl.get(pointer);
+                        mDbHelper.writeToDatabase(imgurl, dbposition, writabledb);
+                        Log.d("dbposition", String.valueOf(dbposition));
+                        dbposition++;
+                        int i = mDbHelper.CountItemsdb(writabledb);
+                        Log.d("Saving image", "URL:"+imgurl+String.valueOf(i));
+
+                    }
+                });
+            }
+            case("previous"):{
+                Button Previous;
+                Previous = (Button) v.findViewById(R.id.prev_image);
+                Previous.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        if(listurl.size()>0){
+                            pointer--;
+                        }
+                        else{
+                            pointer=0;
+                        }
+                        reload_webview();
+                    }
+                });
+            }
+            case("next"):{
+                Button Next;
+                Next = (Button) v.findViewById(R.id.next_image);
+                Next.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        pointer++;
+                        reload_webview();
+                    }
+                });
+            }
+            case("search"):{
+                ImageButton Search;
+                Search = (ImageButton) v.findViewById(R.id.search_button);
+                Search.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        String search = editText.getText().toString();
+                        Log.d("search keyword", search);
+                        google_search(search);
+                    }
+                });
+            }
+        }
+
+    }
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View wholeview = inflater.inflate(R.layout.fragment_image_query, container, false);
+        create_button(wholeview,"save",mDbHelper,writabledb);
+        create_button(wholeview,"previous",mDbHelper,writabledb);
+        create_button(wholeview,"next",mDbHelper,writabledb);
+        create_button(wholeview,"search",mDbHelper,writabledb);
+
```

Comment:
yep, like i said above, you might as well have a create_buttons method that does all of these.